sljit_src/sljitNativeX86_common.c

```diff
@@ -572,12 +572,13 @@ static void get_cpu_features(void)
 			feature_list |= CPU_FEATURE_OSXSAVE;
 		if (info[2] & 0x10000000)
 			feature_list |= CPU_FEATURE_AVX;
-#if (defined SLJIT_DETECT_SSE2 && SLJIT_DETECT_SSE2)
-		if (info[3] & 0x4000000)
-			feature_list |= CPU_FEATURE_SSE2;
-#endif
 		if (info[3] & 0x8000)
 			feature_list |= CPU_FEATURE_CMOV;
+#if (defined(SLJIT_DETECT_SSE2) && SLJIT_DETECT_SSE2) \
+	&& (defined(__SSE2__) || (defined(_WIN64) || (defined(_M_IX86_FP) && _M_IX86_FP >= 2)))
```

Comment:
This will disable SSE2 with SUNPRO, TINYC or other compilers than won't provide the defines.